### PR TITLE
fix(*): fix to allow deployment with salt-ssh.

### PR DESCRIPTION
**Pull request by remichristiaan**
Basically use tplroot in sls files and pass vars used in templates through the context.

I've tested this with salt.minion successfully. Please check if you see any possible issues with this commit. I'm not able to test the other parts (yet) due to time constraints, sorry.

### CODE REVIEW

File: salt/cloud.sls [hunk 1]
```diff
@@ -59,6 +60,8 @@ salt-cloud-{{ cloud_section }}:
     - template: jinja
     - makedirs: True
     - exclude_pat: _*
+    - context:
+          salt_settings: {{ salt_settings|json }}
```
Comment (noelmcloughlin) [inline salt/cloud.sls]: Fix indentation here - it's not consistent.

Comment (noelmcloughlin) [reply salt/cloud.sls]: LGTM but just this change is needed.

**Comment by daks** [comment]
@remichristiaan unfortunately tests are failing https://travis-ci.com/saltstack-formulas/salt-formula/jobs/231096106

**Comment by remichristiaan** [comment]
It seems to fail on the `f_default.conf` where it 'sees' the `file_roots` macro as a string. If I run this version with salt-ssh (py3 on ubuntu 18.04)) e.g. `salt-ssh roster=scan 10.10.10.10 state.apply salt.minion` it works ... so the macro is evaluated as a callable object and not as a string. I have to see if I can schedule time next week to look into this further. Are you sure the tests are okay?

File: salt/cloud.sls (hunk 1 above)
Comment (remichristiaan) [reply salt/cloud.sls]: Good catch ;)

**Comment by daks** [comment]
I didn't specifically verified, but they should because no release can be auto-generated if tests do not pass.

**Comment by remichristiaan** [comment]
Not relevant anymore.

**Pull request closed without merging**